### PR TITLE
Tickets/dm 30719

### PR DESCRIPTION
**Pull request by iagaponenko**

### CODE REVIEW

**Review by jgates108 — APPROVED** [review]
Looks good, only minor comments.

File: src/replica/Configuration.h [hunk 1]
```diff
@@ -448,6 +451,45 @@ class Configuration {
...
+     *   system.
+     * @param workerIsReadOnly This flag will be considered only if 'isEnabled' is set
+     *   to 'true'. The flag narrows down a subset of the 'enabled' workers which
+     *   are either the read-only sources (if 'isReadOnly' is set to true')
+     *   or the read-write replica sources/destinations.
+     * @return Return the effective level for the family given the above-explained
+     *   restrictions.
+     * @throw std::invalid_argument If the empty string passed as a value of the family,
+     *   or if the specified entry was not found in the configuration.
+     */
```
[the remark this answers is not in the document]
Comment (iagaponenko) [reply src/replica/Configuration.h:479]: Workers can be in 3 states:
- `READ-WRITE` (be both a source and a destination for replicas)
- `READ-ONLY` (could only be used as a source of the replicas, but not for accommodating any new replicas)
- `DISABLED` (excluded from any replication operations)

These two boolean flags are meant to express these states:
```
READ-WRITE == isEnabled && !isReadOnly 
READ-ONLY == isEnabled && isReadOnly
DISABLED == !isEnabled
```
The combinatorics of these flags may be a source of confusion. I must admit, this is not the first time someone is asking me about their _exact_ meaning. I guess the right approach here would be re replace these flags with a single parameter of the following type:
```
enum WorkerStatusSelector {
    READ_WRITE,   ///< Is be both a source and a destination for replicas.
    READ_ONLY,    ///< Could only be used as a source of the replicas, but not for accommodating any new replicas.
    DISABLED      ///< Is completely excluded from any replication operations
};
```

File: src/replica/ConfigurationSchema.cc [hunk 2]
```diff
@@ -134,6 +134,13 @@ json const ConfigurationSchema::_schemaJson = json::object(
             {{"description",
               "A path to a folder where Qserv master stores its empty chunk lists. Must be non-empty."},
              {"default", "/qserv/data/qserv"}}},
+           {"max-repl-level",
+            {{"description",
+              "The maximum replicaton level that applies to any database family. This"
+              " hard limit is meant to prevent the Replication system's algorithms from utilizing too"
+              " much storage in case if some large number was accidentally specified by a data"
+              " administrator. The limit is enforced at run time. The number must be greater than 0."},
```
[the remark this answers is not in the document]
Comment (iagaponenko) [reply src/replica/ConfigurationSchema.cc:142]: 👍

File: src/replica/HttpReplicationLevelsModule.h [hunk 3]
```diff
@@ -65,7 +73,21 @@ class HttpReplicationLevelsModule : public HttpModule {
...
                                 HealthMonitorTask::Ptr const& healthMonitorTask);
 
-    // Input parameters
+    /// @return nlohmann::json The replica status report for all known families.
+    nlohmann::json _get();
+
+    /**
+     * @brief Set the replication level of a family.
+     * @return nlohmann::json The replica status report for all known families.
+     */
```
[the remark this answers is not in the document]
Comment (iagaponenko) [reply src/replica/HttpReplicationLevelsModule.h:82]: 👍

File: src/replica/Configuration.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (iagaponenko) [reply src/replica/Configuration.h:479]: I have analyzed the code and realized that improving the API of the Configuration service should rather be done in a dedicated *focused* ticket. Hence, I have registered the following effort: https://jira.lsstcorp.org/browse/DM-36346

**Pull request merged**